### DISCUSSION TITLE
What is more important, accuracy, or fangame use?

### DISCUSSION THREAD

**Discussion by NeonSRB2** [Techokami/SonicWorldsNext#191] [General]
I've felt divided on this with the changes I made to horizontalLockTimer, but I've ran into another problem with CC that could be a real problem for anyone that uses larger tiles in this engine, so lets explain the original issue, though I'll say that this is using the original Godot 3 Branch.

CC, despite being made in Godot 3, uses larger tile sizes for collision, and therefore a raycast starting inside a tile is unlikely to detect anything. If collision got close to the origin of the raycast, the player could more likely clip through, since no tiles pushed the player back at least far enough for raycasts to detect the tile again if they ever missed. This could not be guaranteed anymore when using larger tile sizes. The normal behavior for jumping upward removes floor collision, so if a slope or ledge was below the player, the surface is supposed to get closer to the player than it normally would if the player was falling, along with ruining any sense of momentum by hitting the wall collision inside.

For CC, and what I can consider many fangames desire is the ability to use large tiles easily. Sonic Worlds Next is also very concerned with accuracy, but Sonic never used tiles that big for collision, raising the question of not supporting such sizes, and maybe reintroducing Chunk Zone to help it. However, I can't imagine fangames not wanting to preserve momentum if they jump into a steep slope, and would perhaps appreciate if the engine could handle _any_ collision setup rather than just 16x16 tilemaps. Same could be said for CC's horizontalLockTimer as well, being more consistent and lenient instead of the more accurately strange implementation we currently have. A similar thing can also be said about Sonic Mania fixing the 1 frame of lag when jumping, or getting rid of roll lock.

So here's the big thing; should we make changes to make fangames easier to make and play, or try to make accuracy more of a concern? Of course these could be put into separate branches, but then it questions which branch will be the main one. I think that since this is a successor to Sonic Worlds, including the planned addition of Shadow, it probably should be focused less on accuracy in comparison to more expected functionality. Though of course, I'm just a contributor, so really it's probably the job of Techokami or DimensionWarped to decide or tell me otherwise. If it is otherwise though... I'll probably do what LakeFeperd did and release a branch focused on user-friendliness.

**Comment by Caverns4** [comment]
Accuracy is a major priority, The default Tile size is 16x16, which matches the original Genesis games, and with the patterns system in Godot 4, you can design levels in a very similar way to the original genesis games without too much hassle.
When making additions ore suggestions towards Next, I recommend keeping the 16x16 tile system in mind as that's what most people will likely use.

**Comment by zFriez** [comment]
Since this discussion is still open, I'll ask something somewhat related. I've been doing some testing with the project and I noticed that the default game resolution is 320x224, which is the original resolution for the Genesis, but I also saw that some games have 320x240 resolution. Why the project chose a 320x224 resolution instead of the other? I'm asking this because I've been looking into implementing some sort of a support for a higher resolution, and for integer scaling the 320x240 resolution seems to be easier to implement than the other one.

**Reply by Caverns4** [reply]
> 320x224, which is the original resolution for the Genesis

Answered your own question there. If you would rather use 320x240 your your own project you can obviously do that, such a change should be trivially simple to do, but in the interest of keeping a Genesis-accurate look and feel, we wanted to keep it faithful to the original.
